brave/brave-core
# WIP: Brave Rewards can now be completely reset.

Fixes brave/brave-browser#1192

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:

1. Start Brave, clean profile and enable Rewards.
2. Perform actions on Rewards (contributions, Ads, etc)
3. Go to Settings->Advanced->Reset Rewards Settings
4. On the dialog click 'Reset Settings'
5. Ensure all files related to Rewards are removed from file system and that all Rewards pages are reverted to their 'pre-opt-in' state.
6. Enable Rewards again.
7. Ensure that a new wallet is created and that all Rewards functions perform correctly.


## Reviewer Checklist:
